tests/5.0/loop/test_loop_bind_device.F90

```diff
@@ -0,0 +1,246 @@
+!===--- test_loop_bind_device.F90 ------------------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test checks the loop directive with the bind(binding) clause. The bind
+! clause indicates that the loop construct should apply in the context of the
+! given binding, one of teams, parallel, or thread. Each of these bindings
+! is tested in an appropriate context and the correctness of results of
+! array operations in the nested loop is checked. This test checks these
+! directives in a target context.
+!
+!//===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 32
+
+PROGRAM test_loop_bind_device
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+
+  OMPVV_TEST_VERBOSE(test_bind_teams() .NE. 0)
+  OMPVV_TEST_VERBOSE(test_bind_parallel() .NE. 0)
+  OMPVV_TEST_VERBOSE(test_bind_thread_teams() .NE. 0)
+  OMPVV_TEST_VERBOSE(test_bind_thread_parallel() .NE. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION test_bind_teams()
+    INTEGER,DIMENSION(N,N):: x
+    INTEGER,DIMENSION(N):: y
+    INTEGER,DIMENSION(N):: z
+    INTEGER:: errors, num_teams, i, j
+
+    OMPVV_INFOMSG("test_loop_bind_teams")
+
+    errors = 0
+    num_teams = -1
+
+    DO i = 1, N
+       DO j = 1, N
+          x(j,i) = 1
+       END DO
+       y(i) = i
+       z(i) = 2*i
+    END DO
+
+    !$omp target teams num_teams(OMPVV_NUM_TEAMS_DEVICE) thread_limit(OMPVV_NUM_THREADS_DEVICE) map(tofrom: x, y, z, num_teams)
+    !$omp loop bind(teams)
```

Comment:
I don't really understand the bind, so I have these questions:
Are we interested on checking the behavior of the binding or just the compiler ability to compile correctly?
If the compiler or runtime ignores the bind(teams), would this produce the correct results?

Reply:
This test just checks whether the compiler can compile correctly or not; even if the compiler ignores the `bind(team)` clause, the correct compiler should generate an output program producing the correct results since the `loop` construct guarantees that each logical iteration is executed once per instance of the `loop` region, no matter how the binding thread set is decided.